libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -552,6 +555,35 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPoseStamped& msg)
 }
 #endif // AP_DDS_GEOPOSE_PUB_ENABLED
 
+#if AP_DDS_GOAL_PUB_ENABLED
+bool AP_DDS_Client::update_topic_goal(geographic_msgs_msg_GeoPointStamped& msg)
+{
+    const auto &vehicle = AP::vehicle();
+    update_topic(msg.header.stamp);
+    Location target_loc;
+    vehicle->get_target_location(target_loc);
+    msg.position.latitude = target_loc.lat * 1e-7;
+    msg.position.longitude = target_loc.lng * 1e-7;
+    msg.position.altitude = target_loc.alt * 1e-2;
```

Comment:
What about the frame? You need to convert the goal target location to absolute before putting it in geopose. 

See https://github.com/ArduPilot/ardupilot/blob/f08a2120a8ce990e3939496dbe50ea1d4cae9fcb/libraries/AP_Common/Location.h#L60C4-L60C51

If the goal is in altitude relative to home, you've now populated the goal with the wrong value.

Reply:
thanks for the suggestion, awesome. Applied.